server/container/feature-checks/src/main/java/org/apache/james/healthcheck/MailReceptionCheck.java

```diff
@@ -0,0 +1,270 @@
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.healthcheck;
+
+import static reactor.core.publisher.Sinks.EmitFailureHandler.FAIL_FAST;
+
+import java.time.Duration;
+import java.time.temporal.ChronoUnit;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.UUID;
+
+import javax.inject.Inject;
+import javax.mail.internet.InternetAddress;
+
+import org.apache.commons.io.IOUtils;
+import org.apache.james.core.Username;
+import org.apache.james.core.builder.MimeMessageBuilder;
+import org.apache.james.core.healthcheck.ComponentName;
+import org.apache.james.core.healthcheck.HealthCheck;
+import org.apache.james.core.healthcheck.Result;
+import org.apache.james.events.Event;
+import org.apache.james.events.EventBus;
+import org.apache.james.events.EventListener;
+import org.apache.james.mailbox.MailboxManager;
+import org.apache.james.mailbox.MailboxSession;
+import org.apache.james.mailbox.MessageManager;
+import org.apache.james.mailbox.events.MailboxEvents.Added;
+import org.apache.james.mailbox.events.MailboxIdRegistrationKey;
+import org.apache.james.mailbox.exception.MailboxException;
+import org.apache.james.mailbox.exception.MailboxNotFoundException;
+import org.apache.james.mailbox.model.FetchGroup;
+import org.apache.james.mailbox.model.MailboxPath;
+import org.apache.james.mailbox.model.MessageRange;
+import org.apache.james.server.core.MailImpl;
+import org.apache.james.user.api.UsersRepository;
+import org.apache.james.util.DurationParser;
+import org.apache.mailet.MailetContext;
+import org.reactivestreams.Publisher;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.github.fge.lambdas.Throwing;
+import com.google.common.base.MoreObjects;
+import com.google.common.collect.ImmutableList;
+
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+import reactor.core.publisher.Sinks;
+import reactor.core.scheduler.Schedulers;
+
+public class MailReceptionCheck implements HealthCheck {
+    public static class Configuration {
+        private static final Duration DEFAULT_TIMEOUT = Duration.ofMinutes(1);
+        public static final Configuration DEFAULT = new Configuration(Optional.empty(), DEFAULT_TIMEOUT);
+
+        public static Configuration from(org.apache.commons.configuration2.Configuration configuration) {
+            Optional<Username> username = Optional.ofNullable(configuration.getString("reception.check.user", null))
+                .map(Username::of);
+            Duration timeout = Optional.ofNullable(configuration.getString("reception.check.timeout", null))
+                .map(s -> DurationParser.parse(s, ChronoUnit.SECONDS))
+                .orElse(DEFAULT_TIMEOUT);
+
+            return new Configuration(username, timeout);
+        }
+
+        private final Optional<Username> testUser;
```

Comment:
I think `checkUser` is better (same name with property in the config file)

Reply:
Mmm maybe other opinions here cc @Arsnael ? 

Check vs test... In this case I prefer the 'test' terminology...

Reply:
If `test` is used, we should change `reception.check.user` -> `reception.test.user`

Reply:
Ok for check user if it please you then.

Reply:
Well I have no issue with that terminology. We use a test user to check that our mailing process works

I would guess `reception.check.user` is more related to the name of the class `MailReceptionCheck`? And we have an other param using this prefix as well. I really don't see anything wrong here :)